flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/v2/PaimonSink.java

```diff
@@ -33,30 +37,35 @@
 import org.apache.paimon.options.Options;
 import org.apache.paimon.table.sink.CommitMessageSerializer;
 
+import java.util.Collection;
+import java.util.UUID;
+
 /**
  * A {@link Sink} for Paimon. Maintain this package until Paimon has it own sinkV2 implementation.
  */
-public class PaimonSink<InputT> implements WithPreCommitTopology<InputT, MultiTableCommittable> {
+public class PaimonSink<InputT>
+        implements WithPreCommitTopology<InputT, MultiTableCommittable>,
+                SupportsWriterState<InputT, PaimonWriterState> {
 
     // provided a default commit user.
     public static final String DEFAULT_COMMIT_USER = "admin";
 
     protected final Options catalogOptions;
 
+    /** The commitUser should be restored in state and restore it in writer. */
     protected final String commitUser;
 
     private final PaimonRecordSerializer<InputT> serializer;
 
     public PaimonSink(Options catalogOptions, PaimonRecordSerializer<InputT> serializer) {
-        this.catalogOptions = catalogOptions;
-        this.serializer = serializer;
-        commitUser = DEFAULT_COMMIT_USER;
+        this(catalogOptions, DEFAULT_COMMIT_USER, serializer);
     }
 
     public PaimonSink(
             Options catalogOptions, String commitUser, PaimonRecordSerializer<InputT> serializer) {
         this.catalogOptions = catalogOptions;
-        this.commitUser = commitUser;
+        // generate a random commit user to avoid conflict.
+        this.commitUser = commitUser + UUID.randomUUID();
```

Comment:
Can we introduce `commit.user-prefix` as an alias of `commit.user`? Paimon also uses this option name, and I think it describes how commitUser is created better.